genshin/client/components/base.py

```diff
@@ -218,6 +218,14 @@ def set_redis_cache(
         redis = aioredis.Redis.from_url(url, **redis_kwargs)  # pyright: ignore[reportUnknownMemberType]
         self.cache = client_cache.RedisCache(redis, ttl=ttl, static_ttl=static_ttl)
 
+    def set_proxy(self, proxy: yarl.URL) -> None:
+        """Set proxy."""
+        self.cookie_manager.proxy = proxy
```

Comment:
Setters should only be used for objects that require multiple arguments, this can just be a property.